README.md

```diff
@@ -385,21 +387,46 @@ There are two built-in casting operators in **Jakt**.
 - `as! T`: Returns a `T`, aborts the program if the source value isn't convertible to `T`.
 
 The `as` cast can do these things (note that the implementation may not agree yet):
-- Casts to the same type are infallible and pointless, so might be forbidden in the future.
-- If both types are primitive, a safe conversion is done.
-    - Integer casts will fail if the value is out of range. This means that promotion casts like i32 -> i64 are infallible.
-    - Float -> Integer casts truncate the decimal point (?)
-    - Integer -> Float casts resolve to the closest value to the integer representable by the floating-point type (?). If the integer value is too large, they resolve to infinity (?)
-    - Any primitive -> bool will create `true` for any value except 0, which is `false`.
-    - bool -> any primitive will do `false -> 0` and `true -> 1`, even for floats.
-- If the types are two different pointer types (see above), the cast is essentially a no-op. A cast to `T` will increment the reference count as expected; that's the preferred way of creating a strong reference from a weak reference. A cast from and to `raw T` is unsafe.
-- If the types are part of the same type hierarchy (i.e. one is a child type of another):
-    - A child can be cast to its parent infallibly.
-    - A parent can be cast to a child, but this will check the type at runtime and fail if the object was not of the child type or one of its subtypes.
-- If the types are incompatible, a user-defined cast is attempted to be used. The details here are not decided yet.
-- If nothing works, the cast will not even compile.
-
-Additional casts are available in the standard library. Two important ones are `as_saturated` and `as_truncated`, which cast integral values while saturating to the boundaries or truncating bits, respectively.
+- [ ] Casts to the same type are infallible and pointless, so might be forbidden in the future.
+- [ ] If both types are primitive, a safe conversion is done.
+    - [ ] Integer casts will fail if the value is out of range. This means that promotion casts like i32 -> i64 are infallible.
+    - [ ] Float -> Integer casts truncate the decimal point, if the float is a normal numeric value. If the float is any NaN or either Infinity, the cast fails.
+    - [ ] Integer -> Float casts resolve to the closest value to the integer representable by the floating-point type. If the integer value is not exactly representable, the cast fails.
+    - [ ] Any primitive -> bool will create `true` for any value except 0, which is `false`.
+    - [ ] bool -> any primitive will do `false -> 0` and `true -> 1`, even for floats.
+- If the types are two different pointer types of the same underlying type `T`, the cast is generally allowed and infallible.
```

Comment:
```
enum Foo: i32 {
    A
    B
}

enum Bar: i32 {
    A = 1
    B
}

Foo::A as! Bar
```

Not safe, definitely not correct, and most certainly not infallible - we don't check them right now, but casting to an enum type should be guaranteed to produce one of the variants.

Reply:
`Foo::A` is an `i32`. `Bar` is not `i32`. So this does not compile, and this is not what I mean with "pointer types", that term is reserved for `class`, `&`, `&mut`. What you're looking at is probably type aliases, but neither Foo nor Bar are a type alias of i32 like `using` would be, as (and you correctly identified this) not all instances of i32 are valid instances of Foo and Bar. It is just the case that instances of Foo and Bar are either of type i32 or of a type that we can implicitly treat as i32 (no clue how *exactly* we currently implement that).

Reply:
Ah, I misread that, my bad.